c-dependencies/js-compute-runtime/js-compute-builtins.cpp

```diff
@@ -4198,23 +4202,39 @@ bool init_class(JSContext *cx, HandleObject global) {
   return !!GET_atom;
 }
 
+JSObject *create(JSContext *cx, HandleObject requestInstance, RequestHandle request_handle,
+                 BodyHandle body_handle, bool is_downstream);
+
 JSObject *create(JSContext *cx, RequestHandle request_handle, BodyHandle body_handle,
                  bool is_downstream) {
-  RootedObject request(cx, JS_NewObjectWithGivenProto(cx, &class_, proto_obj));
-  if (!request)
+  RootedObject requestInstance(cx, JS_NewObjectWithGivenProto(cx, &class_, proto_obj));
+  if (!requestInstance)
     return nullptr;
 
-  JS::SetReservedSlot(request, Slots::Request, JS::Int32Value(request_handle.handle));
-  JS::SetReservedSlot(request, Slots::Headers, JS::NullValue());
-  JS::SetReservedSlot(request, Slots::Body, JS::Int32Value(body_handle.handle));
-  JS::SetReservedSlot(request, Slots::BodyStream, JS::NullValue());
-  JS::SetReservedSlot(request, Slots::HasBody, JS::FalseValue());
-  JS::SetReservedSlot(request, Slots::BodyUsed, JS::FalseValue());
-  JS::SetReservedSlot(request, Slots::Method, JS::StringValue(GET_atom));
-  JS::SetReservedSlot(request, Slots::CacheOverride, JS::NullValue());
-  JS::SetReservedSlot(request, Slots::IsDownstream, JS::BooleanValue(is_downstream));
+  return Request::create(cx, requestInstance, request_handle, body_handle, is_downstream);
+}
+
+JSObject *create(JSContext *cx, HandleObject requestInstance, RequestHandle request_handle,
+                 BodyHandle body_handle, bool is_downstream) {
+  JS::SetReservedSlot(requestInstance, Slots::Request, JS::Int32Value(request_handle.handle));
+  JS::SetReservedSlot(requestInstance, Slots::Headers, JS::NullValue());
+  JS::SetReservedSlot(requestInstance, Slots::Body, JS::Int32Value(body_handle.handle));
+  JS::SetReservedSlot(requestInstance, Slots::BodyStream, JS::NullValue());
+  JS::SetReservedSlot(requestInstance, Slots::HasBody, JS::FalseValue());
+  JS::SetReservedSlot(requestInstance, Slots::BodyUsed, JS::FalseValue());
+  JS::SetReservedSlot(requestInstance, Slots::Method, JS::StringValue(GET_atom));
+  JS::SetReservedSlot(requestInstance, Slots::CacheOverride, JS::NullValue());
+  JS::SetReservedSlot(requestInstance, Slots::IsDownstream, JS::BooleanValue(is_downstream));
 
-  return request;
+  return requestInstance;
+}
+
+JSObject *create(JSContext *cx, HandleValue input, HandleValue init) {
+  RootedObject requestInstance(cx, JS_NewObjectWithGivenProto(cx, &class_, proto_obj));
+  if (!requestInstance)
+    return nullptr;
+
+  return Request::create(cx, requestInstance, input, init);
 }
```

Comment:
Is this `create` method still used? It looks like the same signature as the one that was changed above, but it's hard to tell what might still be using it.

Reply:
Yes, this is used in our fetch implementation - I've changed the code now to not make use of this create method